tests/scripts/run-xunit-perf.py

```diff
@@ -0,0 +1,454 @@
+
+    coreclrRepo = os.getcwd()
+    etwCollection = 'Off' if collectionFlags == 'stopwatch' else 'On'
+    sandboxDir = os.path.join(coreclrRepo, 'bin', 'sandbox')
+    sandboxOutputDir = outputDir if not outputDir is None else os.path.join(sandboxDir, 'Logs')
+
+    extension = '.exe' if platform == 'Windows_NT' else ''
+
+    my_env = os.environ
+    dotnetEnv = os.environ
+
+    dotnetEnv['DOTNET_MULTILEVEL_LOOKUP'] = '0'
+    dotnetEnv['UseSharedCompilation'] = 'false'
+
+    # Setup directories
+    log('Setting up directories')
+    coreclrOverlay = verify_core_overlay(coreclrRepo, platform, arch, configuration)
+    setup_sandbox(sandboxDir)
+
+    os.chdir(sandboxDir)
+
+    perfRunLog = set_perf_run_log(sandboxOutputDir)
+    build_perfharness(coreclrRepo, sandboxDir, extension, dotnetEnv)
+
+    my_env = setup_optimization_level(my_env, optLevel)
+
+    if not 'XUNIT_PERFORMANCE_MAX_ITERATION' in my_env:
+        my_env['XUNIT_PERFORMANCE_MAX_ITERATION'] = '21'
+    if not 'XUNIT_PERFORMANCE_MAX_ITERATION_INNER_SPECIFIED' in my_env:
+        my_env['XUNIT_PERFORMANCE_MAX_ITERATION_INNER_SPECIFIED'] = '21'
+
+    # Copy core overlay contents to sandbox
+    copytree(coreclrOverlay, sandboxDir)
+
+    # Determine benchmarks we will be running
+    benchmarks = []
+    if not sliceNumber == -1:
+        with open(sliceConfigFile) as jsonData:
+            data = json.load(jsonData)
+            if sliceNumber >= len(data["slices"]):
+                raise ValueError('Invalid slice number. %s is greater than the max number of slices %s' % (sliceNumber, len(data["slices"])))
+            for benchmark in data["slices"][sliceNumber]["folders"]:
+                benchmarks += [benchmark]
+
+    else:
+    # If slice was not specified, run everything in the coreclrPerf directory. Set benchmarks to an empty string
+        benchmarks = [{ 'directory' : '', 'extraFlags': ''}]
+
+    testFileExt = 'dll' if isLibrary else 'exe'
+
+    # Run benchmarks
+    failures = 0
+    lvRunId = 'Perf-%s' % etwCollection
+
+    for benchmark in benchmarks:
+        testFileExt = 'dll' if benchmark["extraFlags"] == '-library' else 'exe'
+        benchmarkDir = os.path.normpath(benchmark["directory"])
+
+        testPath = os.path.join(coreclrPerf, benchmarkDir)
+
+        for root, dirs, files in os.walk(testPath):
+            for f in files:
+                if f.endswith(testFileExt):
+                    benchname, ext = os.path.splitext(f)
+
+                    benchmarkOutputDir = os.path.join(sandboxOutputDir, 'Scenarios') if isScenarioTest else os.path.join(sandboxOutputDir, 'Microbenchmarks')
+                    benchmarkOutputDir = os.path.join(benchmarkOutputDir, etwCollection, benchname)
+
+                    failures += run_benchmark(benchname, root, my_env, sandboxDir, benchmarkOutputDir, testFileExt, stabilityPrefix, collectionFlags, lvRunId, etwCollection, isScenarioTest, arch, extension, executable)
```

Comment:
>run_benchmark [](start = 32, length = 13)

If the benchmark failed to run, then we should not try to generate BenchView data (it will most likely fail).

Reply:
OK.